lib/runtime/src/lib.rs

```diff
@@ -1,51 +1,17 @@
-#[cfg(test)]
-#[macro_use]
-extern crate field_offset;
+#[cfg(feature = "wasmer-clif-backend")]
+extern crate wasmer_clif_backend;
 
-#[macro_use]
-mod macros;
-#[doc(hidden)]
-pub mod backend;
-mod backing;
-pub mod error;
-pub mod export;
-pub mod import;
-pub mod instance;
-pub mod memory;
-pub mod module;
-mod sig_registry;
-pub mod structures;
-mod sys;
-pub mod table;
-pub mod types;
-pub mod vm;
-#[doc(hidden)]
-pub mod vmcalls;
+extern crate wasmer_runtime_core;
 
-use self::error::CompileResult;
-#[doc(inline)]
-pub use self::error::Result;
-#[doc(inline)]
-pub use self::instance::Instance;
-#[doc(inline)]
-pub use self::module::Module;
-use std::rc::Rc;
+pub use wasmer_runtime_core::*;
 
-pub mod prelude {
-    pub use crate::import::{ImportObject, Namespace};
-    pub use crate::types::{
-        FuncIndex, GlobalIndex, ImportedFuncIndex, ImportedGlobalIndex, ImportedMemoryIndex,
-        ImportedTableIndex, LocalFuncIndex, LocalGlobalIndex, LocalMemoryIndex, LocalTableIndex,
-        MemoryIndex, TableIndex, Type, Value,
-    };
-    pub use crate::vm;
-    pub use crate::{export_func, imports};
-}
+#[cfg(feature = "wasmer-clif-backend")]
+use wasmer_clif_backend::CraneliftCompiler;
 
-/// Compile a webassembly module using the provided compiler.
-pub fn compile(wasm: &[u8], compiler: &dyn backend::Compiler) -> CompileResult<module::Module> {
-    let token = backend::Token::generate();
-    compiler
-        .compile(wasm, token)
-        .map(|inner| module::Module::new(Rc::new(inner)))
+/// Compiles a WebAssembly module
+pub fn compile(wasm: &[u8]) -> error::CompileResult<module::Module> {
+    #[cfg(not(feature = "wasmer-clif-backend"))]
+    panic!("compile/1 is not available when default compiler is disabled");
```

Comment:
This should use `compile_error!("compile/1 is not available when default compiler is disabled")` instead of panic.

Reply:
Actually, I'm not sure about that. This would prevent it from even compiling without that feature enabled. Perhaps we should only define the compile function when there's a default compiler?

Reply:
I was thinking they’d use compile/2 and provide compiler parameter if default compiler is disabled.  I’m fine with either error at runtime or conditional inclusion of the method. I thought error at runtime may be more convenient.

Reply:
I was thinking they’d use compile/2 and provide compiler parameter if default compiler is disabled.  I’m fine with either error at runtime or conditional inclusion of the method. I thought error at runtime may be more convenient.

Reply:
I changed the name of `compile` in `runtime-core` to `compile_with` and reexported it in `runtime`.